sqlglot/generator.py

```diff
@@ -394,7 +397,31 @@ def partition_sql(self, expression):
         return f"PARTITION({keys})"
 
     def properties_sql(self, expression):
-        return self.properties("WITH", expression)
+        root_expression = expression.copy()
+        with_expression = expression.copy()
+        table_properties_expression = expression.copy()
```

Comment:
i'm not a fan of how this was done.

we should only have root properties and with properties, tblproperties should only be overridden in hive, hive should overwrite with properties or something to be tblproperties

Reply:
also, let's not copy expression n times and remove, can we just create a new root with the appropriate children?

Reply:
So I do something like this now:
```
root_expression = exp.Properties(expressions=root_properties)
with_expression = exp.Properties(expressions=with_properties)
```

The risk I see though is that since we aren't cloning here we are losing any addition information that could be attached to properties that somewhere in later calls would assume exists. There is not an example of this today but wanted to point out that this could create some non-intuitive behavior. Another way to do this is to clone and then replace `expressions` with a list of properties. From your comment I think what I did is what you are wanting but I want to point out that other options too.

Reply:
yea this is what i was looking for, not exactly sure in understanding the case you’re talking about though